Microsoft.Toolkit.Uwp.UI.Controls/InAppNotification/InAppNotification.cs

```diff
@@ -127,10 +148,134 @@ public void Show(DataTemplate dataTemplate, int duration = 0)
         /// </summary>
         public void Dismiss()
         {
+            Dismiss(InAppNotificationDismissKind.Programmatic);
+        }
+
+        /// <summary>
+        /// Dismiss the notification
+        /// </summary>
+        /// <param name="dismissKind">Kind of action that triggered dismiss event</param>
+        private void Dismiss(InAppNotificationDismissKind dismissKind)
+        {
             if (Visibility == Visibility.Visible)
             {
+                _animationTimer.Stop();
+
+                var eventArgs = new InAppNotificationDismissingEventArgs(dismissKind);
+                Dismissing?.Invoke(this, eventArgs);
+
+                if (eventArgs.Cancel)
+                {
+                    return;
+                }
+
                 VisualStateManager.GoToState(this, StateContentCollapsed, true);
-                Dismissed?.Invoke(this, EventArgs.Empty);
+
+                _animationTimer.Interval = TimeSpan.FromMilliseconds(AnimationDuration);
+                _animationTimer.Tick += DismissAnimationTimer_Tick;
+                _animationTimer.Start();
+            }
+        }
+
+        private void UpdateAnimationDuration(int duration)
```

Comment:
What happens if I've already re-templated the control and added custom duration? Or what if I want to have multiple DoubleAnimationUsingKeyFrames with different durations. Seems like this will break re-templated controls.

Reply:
Yes. It's possible. I don't know what is the best strategy to have default animations.

Reply:
I'd try to see if you can use TemplateBinding in the storyboard in the template itself. That should allow you to remove all of this code and have everything in the template.